cmd/daemon.go

```diff
@@ -102,6 +102,70 @@ func getenv(k, d string) string {
 	return d
 }
 
+type AWSCredentials struct {
+	AWS_ACCESS_KEY_ID     string `json:"AWS_ACCESS_KEY_ID"`
+	AWS_DEFAULT_REGION    string `json:"AWS_DEFAULT_REGION"`
+	AWS_SECRET_ACCESS_KEY string `json:"AWS_SECRET_ACCESS_KEY"`
+}
+
+func awsCredentialsSetup() error {
+	cedana_auth_token, ok := os.LookupEnv("CEDANA_AUTH_TOKEN")
+	if !ok {
+		return fmt.Errorf("CEDANA_AUTH_TOKEN not set")
+	}
+	cedana_url, ok := os.LookupEnv("CEDANA_URL")
+	if !ok {
+		return fmt.Errorf("CEDANA_URL not set")
+	}
+
+	// construct + send request
+	url := fmt.Sprintf("%s/streaming/aws/credentials", cedana_url)
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return fmt.Errorf("error creating request: %v", err)
+	}
+	req.Header.Set("Authorization", "Bearer "+cedana_auth_token)
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return fmt.Errorf("error making request: %v", err)
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("error getting AWS credentials: %d", resp.StatusCode)
+	}
+	respBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return fmt.Errorf("error reading response body: %v", err)
+	}
+
+	// unmarshal response
+	var creds AWSCredentials
+	err = json.Unmarshal([]byte(respBody), &creds)
+	if err != nil {
+		log.Err(err).Msg("Error unmarshaling JSON")
+		return err
+	}
+
+	// set env vars
+	err = os.Setenv("AWS_ACCESS_KEY_ID", creds.AWS_ACCESS_KEY_ID)
+	if err != nil {
+		log.Err(err).Msg("Error setting AWS_ACCESS_KEY_ID")
+		return err
+	}
+	err = os.Setenv("AWS_DEFAULT_REGION", creds.AWS_DEFAULT_REGION)
+	if err != nil {
+		log.Err(err).Msg("Error setting AWS_DEFAULT_REGION")
+		return err
+	}
+	err = os.Setenv("AWS_SECRET_ACCESS_KEY", creds.AWS_SECRET_ACCESS_KEY)
+	if err != nil {
+		log.Err(err).Msg("Error setting AWS_SECRET_ACCESS_KEY")
+		return err
+	}
```

Comment:
I see here we are getting the keys from the response but what is the aws setup script for? It looks like it is ran on every start of the cedana daemon but I thought the aws environment variables were being set in propagator and not in the daemon on helm install

Reply:
i currently have it such that AWS creds can be provided locally or through propagator, in case a customer wants to bring their own bucket. the aws setup script handles both types of AWS credentials -- either setting (1) the 3 env vars `AWS_ACCESS_KEY_ID`, `AWS_DEFAULT_REGION`, and `AWS_SECRET_ACCESS_KEY` or (2) the 2 env vars `AWS_CONFIG_FILE` and `AWS_SHARED_CREDENTIALS_FILE`. AWS automatically selects (1) if any of the variables are set (even set to empty string), so it fails if you `export AWS_ACCESS_KEY_ID=` and then try to set (2). the aws setup script has a bit more robust error checking than that. instead, it selects the group that has non-empty variables set, so that the user can more easily switch back and forth between different sets of credentials.

Reply:
Makes sense!